app/models/ordergroup.rb

```diff
@@ -148,6 +148,27 @@ def account_updated
     financial_transactions.last.try(:created_on) || created_on
   end
 
+  def self.sort_by_param(param)
+    sort_param_map = {
+      "name" => "name",
+      "name_reverse" => "name DESC",
+      "members_count" => "count(*)",
+      "members_count_reverse" => "count(*) DESC",
+      "last_user_activity" => "max(users.last_activity)",
+      "last_user_activity_reverse" => "max(users.last_activity) DESC",
+      "last_order" => "max(orders.starts)",
+      "last_order_reverse" => "max(orders.starts) DESC"
+    }
+
+    @ordergroup = case param
+                  when "members_count", "members_count_reverse", "last_user_activity", "last_user_activity_reverse" then @ordergroup.left_joins(:users).group("groups.id")
+                  when "last_order", "last_order_reverse" then @ordergroup.left_joins(:orders).group("groups.id")
```

Comment:
i don't see where `@ordergroup` is set before. i think you missed to adopt this code

Reply:
Thanks, good catch!